filesystem/Filesystem.php

```diff
@@ -164,80 +164,9 @@ public static function isAbsolute($filename) {
-	 * back to the equivalent filesystem record.
-	 *
-	 * @param int $folderID Folder ID to sync along with all it's children
-	 * @param Boolean $syncLinkTracking Determines if the link tracking data should also
-	 *        be updated via {@link SiteTree->syncLinkTracking()}. Setting this to FALSE
-	 *        means that broken links inside page content are not noticed, at least until the next
-	 *        call to {@link SiteTree->write()} on this page.
-	 * @return string Localized status message
+	 * @deprecated since version 4.0
 	 */
 	public static function sync($folderID = null, $syncLinkTracking = true) {
-		$folder = DataObject::get_by_id('Folder', (int) $folderID);
-		if(!($folder && $folder->exists())) $folder = singleton('Folder');
-
-		$results = $folder->syncChildren();
-		$finished = false;
-		while(!$finished) {
-			$orphans = DB::query('SELECT "ChildFile"."ID" FROM "File" AS "ChildFile"
-				LEFT JOIN "File" AS "ParentFile" ON "ChildFile"."ParentID" = "ParentFile"."ID"
-				WHERE "ParentFile"."ID" IS NULL AND "ChildFile"."ParentID" > 0');
-			$finished = true;
-			if($orphans) foreach($orphans as $orphan) {
-				$finished = false;
-				// Delete the database record but leave the filesystem alone
-				$file = DataObject::get_by_id("File", $orphan['ID']);
-				$file->deleteDatabaseOnly();
-				unset($file);
-			}
-		}
-
-		// Update the image tracking of all pages
-		if($syncLinkTracking) {
-			if(class_exists('SiteTree')) {
-
-				// if subsites exist, go through each subsite and sync each subsite's pages.
-				// disabling the filter doesn't work reliably, because writing pages that share
-				// the same URLSegment between subsites will break, e.g. "home" between two
-				// sites will modify one of them to "home-2", thinking it's a duplicate. The
-				// check before a write is done in SiteTree::validURLSegment()
-				if(class_exists('Subsite')) {
-					// loop through each subsite ID, changing the subsite, then query it's pages
-					foreach(Subsite::get()->getIDList() as $id) {
-						Subsite::changeSubsite($id);
-						foreach(SiteTree::get() as $page) {
-							// syncLinkTracking is called by SiteTree::onBeforeWrite().
-							// Call it without affecting the page version, as this is an internal change.
-							$page->writeWithoutVersion();
-						}
-					}
-
-					// change back to the main site so the foreach below works
-					Subsite::changeSubsite(0);
-				}
-
-				foreach(SiteTree::get() as $page) {
-					// syncLinkTracking is called by SiteTree::onBeforeWrite().
-					// Call it without affecting the page version, as this is an internal change.
-					$page->writeWithoutVersion();
-				}
-			}
-		}
-
-		return _t(
-			'Filesystem.SYNCRESULTS',
-			'Sync complete: {createdcount} items created, {deletedcount} items deleted',
-			array('createdcount' => (int)$results['added'], 'deletedcount' => (int)$results['deleted'])
-		);
+		Deprecation::notice("4.0", "Filesystem can no longer be automatically synchronised");
```

Comment:
It was mentioned that it could be done via the default APL, but we would have to implement this in a separate story. The existing system would have to be removed in any case. Talk to @sminnee about this.

The reference to this in the rfc is at https://github.com/silverstripe/silverstripe-framework/issues/3792#issuecomment-69996682

I imagine the apl would need to have some kind of `canSynchronise` and `doSynchronise` api to reverse generate a set of dataobjects from the asset store.